models/contacts.js

```diff
@@ -1,19 +1,126 @@
-// const fs = require('fs/promises')
+const { readFile, writeFile } = require("fs/promises");
+const { v4: uuidv4 } = require("uuid");
+const path = require("path");
+require("colors");
 
-const listContacts = async () => {}
+const baseDir = __dirname;
+const fileName = "contacts.json";
+const contactsPath = path.join(baseDir, fileName);
 
-const getContactById = async (contactId) => {}
+const contactsArray = () => {
+  return readFile(contactsPath, "utf-8")
+    .then((data) => JSON.parse(data))
+    .catch((error) => console.log(error.message.red));
+};
 
-const removeContact = async (contactId) => {}
+const saveContactArray = (contacts) => {
+  writeFile(contactsPath, JSON.stringify(contacts, null, 2), (error) => {
+    if (error) {
+      console.error(error.message.red);
+    }
+  });
+};
 
-const addContact = async (body) => {}
+const listContacts = async () => {
+  try {
+    const contacts = await contactsArray();
+    return contacts;
+  } catch (error) {
+    console.log(error.message.red);
+    throw error;
+  }
+};
 
-const updateContact = async (contactId, body) => {}
+const getContactById = async (contactId) => {
+  try {
+    const contacts = await contactsArray();
+    const index = contacts.findIndex((contact) => contact.id === contactId);
```

Comment:
Może `contacts.find`?

Reply:
hej tam niżej używam zmiennej index do tego aby przesłać ten kontakt dalej `return contacts[index];` jeśli użyje find to wartość index będzie bool i wszystko mi sie wykrzaczy